contracts/stake-cw721/src/contract.rs

```diff
@@ -0,0 +1,1844 @@
+use crate::hooks::{stake_hook_msgs, unstake_hook_msgs};
+#[cfg(not(feature = "library"))]
+use crate::lazy_static::lazy_static;
+use crate::msg::{
+    ExecuteMsg, GetConfigResponse, GetHooksResponse, InstantiateMsg, QueryMsg,
+    StakedBalanceAtHeightResponse, StakedValueResponse, TotalStakedAtHeightResponse,
+    TotalValueResponse,
+};
+use crate::state::{
+    Config, CONFIG, HOOKS, MAX_CLAIMS, NFT_CLAIMS, REWARD_BALANCE, REWARD_CLAIMS,
+    STAKED_NFTS_PER_OWNER, TOTAL_STAKED_NFTS,
+};
+use crate::ContractError;
+use cosmwasm_std::{
+    entry_point, to_binary, Addr, Binary, Deps, DepsMut, Empty, Env, MessageInfo, Response,
+    StdError, StdResult, Uint128,
+};
+use cw2::set_contract_version;
+use cw20::Cw20ReceiveMsg;
+use cw721::Cw721ReceiveMsg;
+use cw721_controllers::NftClaimsResponse;
+use cw_controllers::ClaimsResponse;
+use cw_utils::Duration;
+use std::collections::HashSet;
+use std::convert::{From, TryFrom};
+
+lazy_static! {
+    static ref DEFAULT_ADDR: Addr = Addr::unchecked("123");
+}
+
+const CONTRACT_NAME: &str = "crates.io:stake_cw721";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response<Empty>, ContractError> {
+    let owner = match msg.owner {
+        Some(owner) => Some(deps.api.addr_validate(owner.as_str())?),
+        None => None,
+    };
```

Comment:
Instead of the match statement you can use map.

```suggestion
    let owner = msg.owner.map(|h| deps.api.addr_validate(&h)).transpose()?;
```

Reply:
Cool.